resources/ecr-scan-check.py

```diff
@@ -0,0 +1,204 @@
+        digest=ecr_scanfindings["imageId"]["imageDigest"],
+    )
+
+    return report_url
+
+
+def generate_slack_json(ecr_imagedata, vulnerabilities, report_url):
+    if vulnerabilities:
+        colour = "#FF0000"
+        heading = "ECR vulnerability scan failure"
+        result = "Vulnerabilities found"
+    else:
+        colour = "#00FF00"
+        heading = "Successful ECR vulnerability scan"
+        result = "No vulnerabilities"
+
+    image_name = ecr_imagedata["repositoryName"]
+    image_tag = ecr_imagedata["imageId"]["imageTag"]
+
+    slack_json = [
+        {
+            "fallback": heading,
+            "pretext": heading
+            + " for *<$ATC_EXTERNAL_URL/teams/$BUILD_TEAM_NAME/pipelines/$BUILD_PIPELINE_NAME/jobs/$BUILD_JOB_NAME/builds/$BUILD_NAME|$BUILD_PIPELINE_NAME>*",
+            "fields": [
+                {"title": "Image Name", "value": image_name, "short": "true"},
+                {"title": "Image Tag", "value": image_tag, "short": "true"},
+                {
+                    "title": "Scan Result:",
+                    "value": "<" + report_url + "|" + result + ">",
+                    "short": "false",
+                },
+            ],
+            "color": colour,
+            "footer": "$ATC_EXTERNAL_URL",
+        }
+    ]
+
+    report_dir = "scan-report"
+    if not os.path.exists(report_dir):
+        os.mkdir(report_dir)
+
+    json_path = "./" + report_dir + "/report.json"
+    with open(json_path, "w", encoding="utf-8") as result_file:
+        json.dump(slack_json, result_file, ensure_ascii=False, indent=4)
+
+    result_file.close()
+
+
+if __name__ == "__main__":
+    # Ensure required environment vars are set
+    if (
+        not os.environ.get("AWS_ACCESS_KEY_ID")
+        or not os.environ.get("AWS_SECRET_ACCESS_KEY")
+        or not os.environ.get("AWS_REGION")
+    ):
+        logoutput("Error", "Required AWS environment variables missing or not set")
+        sys.exit(1)
+
+    # Parse command line arguments/options
+    parser = argparse.ArgumentParser(description="Program arguments and options")
+    parser.add_argument("imagerepo", help="Image repository name")
+    parser.add_argument("imagetag", help="Image tag to filter for")
+    args = parser.parse_args()
+
+    # Determine which severity of vulnerabilities we care about
+    if os.environ.get("MIN_SEVERITY"):
+        severities = get_severities(os.environ.get("MIN_SEVERITY"))
+    else:
+        severities = get_severities("ANY")
+
+    logoutput("Info", "Image name: " + args.imagerepo + ", Tag: " + args.imagetag)
+
+    # Initialise the boto3 session and return an ECR client
+    ecr_client = ecr_open_session()
+
+    # Query the ECR API to return the image scan findings
+    ecr_imagedata = ecr_describe_scan_findings(ecr_client, args.imagerepo, args.imagetag)
+
+    # Wait and try again if the scan status is not COMPLETE
+    while ecr_imagedata["imageScanStatus"]["status"] != "COMPLETE":
```

Comment:
I'd suggest a `MAX_TRY` or `MAX_WAIT` to guard against jobs running indefinitely